### ISSUE TITLE
Not able to retrieve content of messages and attachment

### ISSUE THREAD

**Issue by ritika0817** [microsoftgraph/msgraph-sdk-java#1075]
### Expected behavior

Should return inputStream

### Actual behavior

returns 0 bytes.

### Steps to reproduce the behavior
  
graphClient.users("userEmailId").messages(
        "message_id")
        .content().buildRequest()
        .get();

final InputStream result = new FileAttachmentStreamRequestBuilder(
        graphClient.users("userEmailId").messages("msgId")
            .attachments("attachmentId")
            .buildRequest().getRequestUrl()
            .toString() + "/$value", graphClient, null).buildRequest().get();

Both methods were earlier retrieving content but now I am getting 0 bytes from both methods.

**Comment by ramsessanchez** [comment]
Hi @ritika0817, would you mind letting me know what version of the SDK you are using. I will do my best to reproduce this issue and see why it is that you are not receiving your information, thanks!

**Comment by ritika0817** [comment]
Hi @ramsessanchez
 I am using 5.17.0 version of microsoft-graph.

**Comment by ritika0817** [comment]
<img width="575" alt="zeroBytes" src="https://user-images.githubusercontent.com/105796654/169463236-d636deaa-b3f9-4fb3-af55-083927b3617c.png">

**Comment by ritika0817** [comment]
Is there any update on the ticket?

**Comment by ramsessanchez** [comment]
Hi @ritika0817, sorry for the delay, but I was unable to replicate the error. I am using graph 5.0.22 and successfully retrieved a file attachment using this:
Attachment attachment = graphClient.me().messages("messageId").attachments("attachmentId")
	.buildRequest()
	.get();
I would suggest using the graph explorer (https://developer.microsoft.com/en-us/graph/graph-explorer) to test whether your request would go through and return the correct information using this request: https://graph.microsoft.com/v1.0/me/messages/<messageID>/attachments/<AttachmentID>
Sorry I couldn't reproduce the error, let me know if you run into the same problem while using the graph explorer and share your response error. Thanks!

**Comment by ritika0817** [comment]
The issue is not with getting attachment but with the content of attachment and with graph explorer, I am able to retrieve content but the issue is with sdk method:
final InputStream result = new FileAttachmentStreamRequestBuilder(
graphClient.users("userEmailId").messages("msgId")
.attachments("attachmentId")
.buildRequest().getRequestUrl()
.toString() + "/$value", graphClient, null).buildRequest().get();

**Comment by ramsessanchez** [comment]
Hi @ritika0817, apologies, it seems I misunderstood your question. If you are looking to get the content bytes of the attachments, which you can then convert using a FileOutputStream I suggest this: 
``` Java
FileAttachment attachment = (FileAttachment)graphClient.me().messages(messageID).attachments(attachmentID).buildRequest().get();
byte[] content = attachment.contentBytes;
```

**Comment by ramsessanchez** [comment]
@ritika0817, to answer your question about ItemAttachment, it extends Attachment, but it also has a navigation property to an OutlookItem which is an abstract class. In order to really get the content, you're looking for you would need to know the type of OutlookItem you are dealing with. 
To go back to your earlier code I believe it should work if done in this manner, I have ran it and am able to successfully retrieve the bytes: 
``` Java 
        URL requestUrl = graphClient.me()
                .messages(messageID)
                .attachments(attachmentID)
                .buildRequest().getRequestUrl();
        final InputStream result = new FileAttachmentStreamRequestBuilder(requestUrl.toString()+"/$value", graphClient, null).buildRequest().get();
        byte[] bytes = result.readAllBytes() //bytes should now contain all the info from the stream
```

**Comment by ramsessanchez** [comment]
![stream](https://user-images.githubusercontent.com/63934382/171930421-1d07393c-6d61-4a62-bb11-a9262d19e961.png)

**Comment by ritika0817** [comment]
Thanks @ramsessanchez.... I'll cross check and update you.

**Comment by ritika0817** [comment]
Hi, I am getting content bytes but it injects junk data like 
<img width="252" alt="image" src="https://user-images.githubusercontent.com/105796654/172334640-786dbcfd-4a0c-4520-a6be-33932960ad87.png">
where as the content of the file was ABCD
<img width="553" alt="image" src="https://user-images.githubusercontent.com/105796654/172334804-418f3b09-9293-470d-ad77-e7395b8c1a90.png">

**Comment by ramsessanchez** [comment]
Hi @ritika0817, apologies for the delay. I've ran a sample and did not run into the same issues unfortunately. Following the code from the time before I added:
``` Java
        try{
            OutputStream os = new FileOutputStream("doc.docx");
            os.write(bytes);
            os.close();
        }
        catch (Exception e) { }
```
When I opened the file I did not receive the same junk data issue. Perhaps you have used the wrong file extension? 
<img width="295" alt="image" src="https://user-images.githubusercontent.com/63934382/173931230-b419412f-5bc2-47eb-86a9-69690cce1114.png">
